### PR TITLE
[NEXUS-8304,NEXUS-8231] Add bean mapping for configuration and validation support

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8304
https://issues.sonatype.org/browse/NEXUS-8231

### CODE REVIEW

**Comment by jdillon** [comment]
Also added FacetSupport.facet(), since getRepository().facet() was/is used all over, can reduce a tiny bit of verbosity at the expense of a tiny helper overhead.

FTR I don't think we want to add a lot of these, NX2 did that and ended up with a lot of this sort of thing, that isn't the intent here, but clearly facet lookup is common so I think it makes sense to simplify for this specifically.

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF150-2 :white_check_mark:

**Comment by jdillon** [comment]
Will probably have to sort out how violations of separate facets gets aggregated together with proper keys to apply to the UI...

**Comment by adreghiciu** [comment]
+1

**Comment by mrprescott** [comment]
+1

**Comment by kellyrob99** [comment]
+1

**Comment by cstamas** [comment]
+1

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java [hunk 1]
```diff
@@ -58,19 +60,36 @@
...
 
-  private ORID bucketId;
+  @VisibleForTesting
+  static class Config
+  {
+    @NotEmpty
+    public String blobStoreName = "default";
 
-  private WritePolicy writePolicy;
+    //@NotNull
```
Comment (kellyrob99) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java]: ?

Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java]: This is nullable atm, though probably should not be.

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**